packages/dds/tree/src/simple-tree/schemaFactoryRecursive.ts

```diff
@@ -192,11 +40,57 @@ export function createFieldSchemaUnsafe<
 }
 
 /**
- * Extends SchemaFactory with utilities for recursive types.
+ * Extends SchemaFactory with utilities for recursive schema.
+ *
+ * @remarks
+ * This is separated from SchemaFactory as these APIs are more experimental, and may be stabilized independently.
+ *
+ * The non-recursive versions of the schema building methods will run into several issues when used recursively.
+ * Consider the following example:
  *
- * @remarks This is separated from SchemaFactory as these APIs are more experimental, and may be stabilized independently.
+ * ```typescript
+ * const Test = sf.array(Test); // Bad
+ * ```
  *
- * @sealed @internal
+ * This has several issues:
+ *
+ * 1. It is a structurally named schema.
+ * Structurally named schema derive their name from the names of their child types, which is not possible when the type is recursive since its name would include itself.
+ * Instead a name must be explicitly provided.
+ *
+ * 2. The schema accesses itself before its defined.
+ * This would be a runtime error if the TypeScript compiler allowed it.
+ * This can be fixed by wrapping the type in a function, which also requires explicitly listing the allowed types an an array (`[() => Test]`)
+ *
+ * 3. TypeScript fails to infer the recursive type, and fall back to any with the warning or error (depending on the compiler configuration):
+ * `'Test' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.ts(7022)`.
+ * This issue is what the specialized recursive schema building methods fix.
+ * This fix comes at a cost: to make the recursive cases work, the `extends` clauses had to be removed.
+ * This means that mistakes declaring recursive schema often don't give compile errors in the schema.
+ * Additionally support for implicit construction had to be disabled.
+ * This means that new nested {@link Unhydrated} nodes can not be created like `new Test([[]])`.
+ * Instead the nested nodes must be created explicitly using the construction like`new Test([new Test([])])`.
+ *
+ * 4. It is using "POJO" mode since its not explicitly declaring a new class.
```

Comment:
```suggestion
 * 4. It is using "POJO" mode since it's not explicitly declaring a new class.
```

Also, we should define "POJO" before using it.